src/ducks/account/AccountSwitch.jsx

```diff
@@ -238,139 +268,116 @@ AccountSwitchMenu.propTypes = {
-  return (
-    <div
-      className={cx(styles['account-switch'], {
-        [styles['AccountSwitch--small']]: small
-      })}
-    >
-      {children}
-    </div>
-  )
+const AccountSwitchWrapper = ({ children }) => {
+  return <div className={cx(styles['account-switch'])}>{children}</div>
 }
 
 const barItemStyle = { overflow: 'hidden', paddingRight: '1rem' }
 
+const selectPropsBySize = {
+  normal: {
+    typographyProps: {
+      variant: 'body1',
+      color: 'primary'
+    }
+  },
+  large: {
+    typographyProps: {
+      variant: 'h4',
+      color: 'primary'
+    },
+    arrowProps: {
+      size: 16
+    }
+  },
+  small: {
+    typographyProps: {
+      variant: 'caption',
+      color: 'primary'
+    },
+    arrowProps: {
+      size: 10
+    }
+  }
+}
+
 // Note that everything is set up to be able to combine filters (even the redux store).
 // It's only limited to one filter in a few places, because the UI can only accomodate one right now.
-class AccountSwitch extends Component {
-  state = {
-    open: false
-  }
+const AccountSwitch = props => {
+  const {
+    filteringDoc,
+    filterByDoc,
+    filteredAccounts,
+    resetFilterByDoc,
+    size,
+    accounts: accountsCollection,
+    groups: groupsCollection,
+    virtualGroups
+  } = props
+  const [open, setOpen] = useState()
+  const { t } = useI18n()
+  const { isMobile } = useBreakpoints()
 
-  close = () => {
-    if (this.state.open) {
-      this.setState({ open: false })
-    }
-  }
+  const handleToggle = useCallback(() => setOpen(open => !open), [setOpen])
+  const handleClose = useCallback(() => setOpen(false), [setOpen])
 
-  toggle = () => {
-    let newState = !this.state.open
-    this.setState({
-      open: newState
-    })
-  }
+  const accounts = accountsCollection.data
+  const groups = [...groupsCollection.data, ...virtualGroups].map(group => ({
+    ...group,
+    label: getGroupLabel(group, t)
+  }))
 
-  accountExists = accountId => {
-    const accounts = this.props.accounts.data
-    return accounts.find(account => account.id === accountId)
+  const closeAfterSelect = cb => param => {
+    cb(param)
+    handleClose()
   }
 
-  render() {
-    const {
-      t,
-      filteringDoc,
-      filterByDoc,
-      filteredAccounts,
-      resetFilterByDoc,
-      breakpoints: { isMobile },
-      small,
-      accounts: accountsCollection,
-      groups: groupsCollection,
-      virtualGroups
-    } = this.props
-    const { open } = this.state
-
-    const accounts = accountsCollection.data
-    const groups = [...groupsCollection.data, ...virtualGroups].map(group => ({
-      ...group,
-      label: getGroupLabel(group, t)
-    }))
-
-    if (!accounts || accounts.length === 0) {
-      return isMobile ? (
+  // It seems there is a bug in cozy-client when we delete a document
+  // The document is removed in the store, but still referenced in the collection
+  // So we may get an undefined group. We filter it before sorting
```

Comment:
Do you have created the issue on the cozy-client's side ?

Reply:
This is not a comment from this PR, this surfaces only because the lines have been moved. I will check if an issue is created though, thanks.